test/integration/targets/filter_core/tasks/password_hash.yml

```
@@ -87,3 +87,58 @@
       assert:
         that:
           - yescrypt_hash_salt == '$y$j9T$V7qMYJaNbVKOeh4PhtqPk/$x7rTqZ.RpI07.dkBSxcg.jLM8ODUfx25rCN0cFsAUg0'
+
+    - name: Test salt with invalid character
+      set_fact:
+        md5_hash: "{{ 'test' | password_hash('md5_crypt', salt='123^') }}"
+      ignore_errors: yes
+      register: bad_salt_result
+
+    - name: Determine if bad salt character was recognized
+      assert:
+        that:
+          - "'invalid characters in salt' in bad_salt_result.msg"
```

Assertions like this one can be rewritten to avoid the extra quotes. The list can also be eliminated for a more compact assertion:

```suggestion
        that: bad_salt_result.msg is contains 'invalid characters in salt'
```

---

*(alternatively, I also like using `>-` in many places for the same reason)*